test/NuGet.Core.Tests/NuGet.ProjectModel.Test/PackageSpecReaderTests.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Linq;
+using NuGet.LibraryModel;
+using NuGet.Versioning;
+using Xunit;
+
+namespace NuGet.ProjectModel.Test
+{
+    public class PackageSpecReaderTests
+    {
+
+        [Fact]
+        public void PackageSpecReader_PackageMissingVersion()
+        {
+            // Arrange
+            var json = @"{
+                          ""dependencies"": {
+                                ""packageA"": {
+                                    ""type"": ""build""
+                                }
+                            },
+                            ""frameworks"": {
+                                ""net46"": {}
+                            }
+                        }";
+
+            // Act
+            Exception exception = null;
+
+            try
+            {
+                var spec = JsonPackageSpecReader.GetPackageSpec(json, "TestProject", "project.json");
+            }
+            catch (Exception ex)
+            {
+                exception = ex;
+            }
+
+            // Assert
+            Assert.Contains("specify a version range", exception.Message);
+        }
+
+        [Fact]
+        public void PackageSpecReader_ProjectMissingVersion()
+        {
+            // Arrange
+            var json = @"{
+                          ""dependencies"": {
+                                ""packageA"": {
+                                    ""target"": ""project""
+                                }
+                            },
+                            ""frameworks"": {
+                                ""net46"": {}
+                            }
+                        }";
+
+            // Act
+            var spec = JsonPackageSpecReader.GetPackageSpec(json, "TestProject", "project.json");
+            var range = spec.Dependencies.Single().LibraryRange.VersionRange;
+
+            // Assert
+            Assert.Equal(VersionRange.All, range);
+        }
+
+        [Fact]
+        public void PackageSpecReader_PackageEmptyVersion()
+        {
+            // Arrange
+            var json = @"{
+                          ""dependencies"": {
+                                ""packageA"": {
+                                    ""target"": ""package"",
+                                    ""version"": """"
```

Comment:
Good call, I've added a test. VersionRange.Parse throws the exception there, which is then turned into a FileFormatException. Slightly different code path, but it is more useful for this and correct.